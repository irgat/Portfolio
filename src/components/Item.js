import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
    id: PropTypes.any.isRequired,
    imagePath: PropTypes.string.isRequired,
    isSelected: PropTypes.bool.isRequired,
    onClick: PropTypes.func.isRequired,
    onMouseOver: PropTypes.func,
    onMouseOut: PropTypes.func,
};

const defaultProps = {
    onMouseOver: () => { },
    onMouseOut: () => { },
};

const Item = (props) => {
    return <img src={props.imagePath} alt="" id={props.id} onClick={props.onClick} onMouseOver={props.onMouseOver} onMouseOut={props.onMouseOut} />;
};

Item.propTypes = propTypes;
Item.defaultProps = defaultProps;

export default Item;