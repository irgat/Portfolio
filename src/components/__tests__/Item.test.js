import React from 'react';
import Enzyme from 'enzyme';
import Item from '../Item';

const data = [
    {
        id: '#',
        imagePath: 'folderName/fileName.ext',
        isSelected: true,
        onClick: jest.fn(),
    },
    {
        id: '##',
        imagePath: 'folderName/fileName.ext',
        isSelected: false,
        onClick: jest.fn(),
    },
];

const item = (props) => {
    return Enzyme.shallow(<Item {...props} />);
};

describe.each(data)('<Item />', (props) => {
    it('renders an <img>', () => {
        const wrapper = item(props).find('img');
        expect(wrapper.getElements()).toMatchSnapshot();

        wrapper.simulate('click');
        expect(props.onClick.mock.calls.length).toBe(1);
    });
});
