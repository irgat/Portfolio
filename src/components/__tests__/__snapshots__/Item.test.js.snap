// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> renders an <img> 1`] = `
Array [
  <img
    alt=""
    id="#"
    onClick={[MockFunction]}
    src="folderName/fileName.ext"
  />,
]
`;

exports[`<Item /> renders an <img> 2`] = `
Array [
  <img
    alt=""
    id="##"
    onClick={[MockFunction]}
    src="folderName/fileName.ext"
  />,
]
`;
